opencl-source-provider.cmake

```diff
@@ -10,7 +10,7 @@ if(DEBUG_KERNELS)
   )
 endif()
 
-set(CL_SRC_FILE ${BDA_DIR}/clSources.cpp)
+set(CL_SRC_FILE ${PROJECT_SOURCE_DIR}/${BDA_DIR}/clSources.cpp)
```

Comment:
Still not the build directory. Please use ${PROJECT_BINARY_DIR}.

Reply:
Hello Markus. I'm not sure if I'm following what you want here. Are you saying I should put the `clSources.cpp` elsewhere? Where would that be?

Reply:
I tried using (without thinking much, I confess) `${PROJECT_SOURCE_DIR}` as you suggested, but the value of `${PROJECT_SOURCE_DIR}` gets appended to the elements of the `MAIN_SOURCE_FILES` list, so the build process ends up looking for `clSources.cpp` in a directory that doesn't exist.

Reply:
If you generate the file below `${PROJECT_BINARY_DIR}` then we need to append that source file with complete path to 
the variable opm_simulators_SOURCES. The convention in OPM is to do that (and the generation of the source file) in the macro sources_hook which then should look like:
```
macro (sources_hook)
  if(OPENCL_FOUND)
    include(opencl-source-provider)
    list(APPEND opm-simulators_SOURCES ${PROJECT_BINARY_DIR}/clSources.cpp)
  endif()
endmacro (sources_hook)
```
With this there will be no changes to CMakeLists_files.cmake in this PR.

Sorry for not being clear previously. (Admittedly, I always have to look this procedure up in the code).

Reply:
Got it. Just fixed it.